### PR TITLE
[ui] add support for selecting multiple nodes at once

### PR DESCRIPTION
**Pull request by ChemicalXandco**
## Description

The ability to select multiple nodes at once improves the efficiency of using the graph editor. I have tried to make the node selection behave similar to blender. The box select widget is used by dragging with the left mouse button. The main selected node is slightly brighter than the other selected nodes and has a blue border.
Screenshot_20210115_202417
Screenshot_20210115_193313

## Features list

- select multiple nodes at once
- use of control key modifier to select multiple nodes
- use of box select widget to select multiple nodes
- duplicate selected nodes
- remove selected nodes
- delete data of selected nodes
- move selected nodes together

(Resolves some points on #269)

### CODE REVIEW

**Comment by fabiencastan** [comment]
I often get this error when manipulating nodes in the GraphEditor:
```
Traceback (most recent call last):
  File "C:\dev\meshroom\meshroom\common\qt.py", line 190, in remove
    raise ValueError("QObjectListModel.remove(obj) : obj not in list")
ValueError: QObjectListModel.remove(obj) : obj not in list
[2021-01-22 23:22:10,394][WARNING] file:///C:/dev/meshroom/meshroom/ui/qml/GraphEditor/GraphEditor.qml:446: Error: QObjectListModel.remove(obj) : obj not in list
```

File: meshroom/ui/commands.py [hunk 1]
```diff
@@ -196,10 +206,21 @@ def redoImpl(self):
         self.duplicates = [n.name for n in duplicates]
         return duplicates
 
-    def undoImpl(self):
-        # delete all the duplicated nodes
-        for nodeName in self.duplicates:
-            self.graph.removeNode(nodeName)
+
+class DuplicateNodeListCommand(_DuplicateNodes):
```
Comment (fabiencastan) [inline meshroom/ui/commands.py:210]: Don't need to have 2 classes to do it with one of multiple nodes and inheritance, etc.
Either we keep `DuplicateNodeListCommand` and always use it (even for a single node) or keep `DuplicateNodeCommand` and group the commands for multiple nodes.

**Review by fabiencastan — CHANGES_REQUESTED** [review]
Thanks for the really cool PR!
I have added few notes to clean the code before integration.

**Comment by fabiencastan** [comment]
Small behavior problem in this configuration (duplicate nodes and nodes behind):
image

It gives:
image

While the simple duplicate nodes:
image

Gives the correct result but with a strange selection:
image

**Comment by fabiencastan** [comment]
It would be great the replace the "Alt+click" to select all the nodes after (instead of duplicating the nodes).
So if we want to move a node and all nodes connected behind it (a fairly common need), it would be simple.

**Comment by fabiencastan** [comment]
Regarding the selection, I think we should clear the selection and select all the newly created nodes.

**Comment by fabiencastan** [comment]
Really great PR, this is a life changer!
Looking forward to be able to merge it.

**Comment by ChemicalXandco** [comment]
> Small behavior problem in this configuration (duplicate nodes and nodes behind)

I agree that this does not behave right. Should I remove the options '_ from here' or should I rework them to include the other selected nodes? I think that removing those options and relying on alt + click to select the following nodes is more intuitive than having these options that operate on some nodes that are **not** selected. What do you think @fabiencastan?

**Comment by fabiencastan** [comment]
Good question. I agree that's redundant and makes two ways to do the same.
But, on the other hand, I think it's still useful to have the "from here" option as it's visual and do not require to know the keyboard shortcuts.
Also alt/shift/ctrl can be problematic in remote, or on some linux alt could be used by the os to move the window, etc.

**Comment by natowi** [comment]
Love it :)

demo

Only a few minor layout issues https://github.com/alicevision/meshroom/issues/1389

**Comment by fabiencastan** [comment]
@ChemicalXandco Is it ready for review? Or are you still working on it?

**Comment by ChemicalXandco** [comment]
It is ready

**Review by fabiencastan — APPROVED** [review]

**Comment by fabiencastan** [comment]
Great work!

**Pull request merged**